packages/lesswrong/components/forumEvents/ForumEventCommentForm.tsx

```diff
@@ -0,0 +1,214 @@
+    cursor: "pointer",
+    fontSize: 14,
+    fontWeight: 600,
+    width: "fit-content",
+    marginLeft: "auto",
+    '&:hover': {
+      opacity: 0.7
+    }
+  }
+});
+
+const ForumEventCommentForm = ({
+  open,
+  comment,
+  forumEventId,
+  post,
+  refetch,
+  onClose,
+  anchorEl,
+  classes,
+}: {
+  open: boolean;
+  comment: ShortformComments | null;
+  forumEventId: string;
+  anchorEl: HTMLElement | null;
+  post: PostsMinimumInfo;
+  onClose: () => void;
+  refetch: () => Promise<void>;
+  classes: ClassesType<typeof styles>;
+}) => {
+  const { CommentsNewForm, LWPopper, ForumIcon, CommentsEditForm, CommentBody } = Components;
+
+  const isDesktop = useIsAboveBreakpoint("md")
+
+  const [editFormOpen, setEditFormOpen] = useState(false);
+  const { flash } = useMessages();
+  const updatePopperRef = useRef<(() => Promise<Partial<State>>) | undefined>(undefined);
+
+  const onSubmit = useCallback(async () => {
+    flash("Success! Open the results to view everyone's votes and comments.")
+    await refetch();
+    onClose();
+  }, [flash, onClose, refetch])
+
+  useEffect(() => {
+    const updatePopperPos = () => {
+      void updatePopperRef.current?.();
+    };
+
+    // Backup to prevent it from sticking in the wrong place forever
+    const intervalId = setInterval(() => {
+      updatePopperPos();
+    }, 3000);
+
+    document.addEventListener('pointerup', updatePopperPos);
+
+    return () => {
+      clearInterval(intervalId);
+      document.removeEventListener('pointerup', updatePopperPos);
+    };
+  }, []);
+
+  if (!open || !anchorEl?.isConnected || !isDesktop) {
+    return null;
+  }
+
+  const debateWeekLink = comment ? commentGetPageUrlFromIds({postId: comment.postId, commentId: comment._id}) : postGetPageUrl(post)
+
+  return (
+    <LWPopper open={open} anchorEl={anchorEl} placement="bottom" allowOverflow={false} updateRef={updatePopperRef}>
+      <div className={classes.popperContent}>
+        <div className={classes.triangle}></div>
+        <ForumIcon icon="Close" className={classes.closeIcon} onClick={onClose} />
+        <div className={classes.title}>What made you vote this way?</div>
+        <div>
+          Your response will appear as a comment on{" "}
+          <Link to={debateWeekLink} target="_blank" rel="noopener noreferrer" className={classes.link}>
+            this Debate Week post
+          </Link>
+          , and show next to your avatar on this banner.
+        </div>
+        {!comment && !editFormOpen && (
+          <CommentsNewForm
+            type="reply"
+            post={post}
+            enableGuidelines={false}
+            cancelCallback={() => onClose()}
+            successCallback={onSubmit}
+            prefilledProps={{
+              forumEventId,
+            }}
+            // TODO change submit styling (actually punt, but note in PR)
+            className={classes.commentForm}
+          />
+        )}
+        {comment && !editFormOpen && (
```

Comment:
A minor thing to improve here is the loading state when you edit a comment (it currently goes black for a second while the comment refetches)